assignments/closure.js

```diff
@@ -4,22 +4,48 @@
 // that manipulates variables defined in the outer scope.
 // The outer scope can be a parent function, or the top level of the script.
 
+let numbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+function calculator(arr) {
+  let random = arr[Math.floor(Math.random() * arr.length)];
 
-/* STRETCH PROBLEMS, Do not attempt until you have completed all previous tasks for today's project files */
+  return function squareNums() {
+    let square = random * random;
+    console.log(`The square of ${random} is ${square}`);
+    return square;
+  };
+}
+
+calculator(numbers)();
 
+/* STRETCH PROBLEMS, Do not attempt until you have completed all previous tasks for today's project files */
 
 // ==== Challenge 2: Implement a "counter maker" function ====
-const counterMaker = () => {
-  // IMPLEMENTATION OF counterMaker:
-  // 1- Declare a `count` variable with a value of 0. We will be mutating it, so declare it using `let`!
-  // 2- Declare a function `counter`. It should increment and return `count`.
-  //      NOTE: This `counter` function, being nested inside `counterMaker`,
-  //      "closes over" the `count` variable. It can "see" it in the parent scope!
-  // 3- Return the `counter` function.
+// IMPLEMENTATION OF counterMaker:
+// 1- Declare a `count` variable with a value of 0. We will be mutating it, so declare it using `let`!
+// 2- Declare a function `counter`. It should increment and return `count`.
+//      NOTE: This `counter` function, being nested inside `counterMaker`,
+//      "closes over" the `count` variable. It can "see" it in the parent scope!
+// 3- Return the `counter` function.
+const counterMaker = limit => {
+  let count = 0;
+  return function counter() {
```

Comment:
we may have a parametric counterMaker that takes in the limit and uses a while loop to count rather than have to call the function each time.